students/daniel_grubbs/lesson05/mailroom3.py

```diff
@@ -0,0 +1,172 @@
+            print_donor_list()
+            print()
+        elif full_name == 'menu':
+            return
+        else:
+            break
+
+    # Enter a donation amount
+    while True:
+        try:
+            donation = int(input("Please enter a donation amount. 'menu' to return to original menu: "))
+            if donation == 'menu':
+                return
+            else:
+                break
+        except ValueError:
+            print('Please enter a valid amount.')
+            # continue
+
+    # Enter a new donor
+    donor = get_donor(full_name)
+    if donor is None:
+        donor = full_name
+        donors[donor] = []
+
+    donors[donor].append(donation)
+
+    # Write a thank you for the donor
+    print(letter(donor))
+    # print('{}, Thank you for your donation in the amount of ${:.2f}'.format(full_name, donation))
+    # send_letter_file()
+
+
+def create_report():
+    """Function for creating a report of donors with donation amounts."""
+    donations = []
+
+    print("{:26s} | {:13s} | {:9s} | {:13s}".format("Donor name", "Total Donation", "Number of Gifts", "Average Gifts"))
+    print("-" * 80)
+
+    for donor, gift in donors.items():
+        total_given = sum(gift)
+        number_gifts = len(gift)
+        average_gift = total_given / number_gifts
+        donations.append((donor, total_given, number_gifts, average_gift))
+
+    for amount in donations:
+        print("{:26s} | {:14.2f} | {:15d} | {:13.2f}".format(*amount))
+    print()
+
+
+def letter(donor):
+    """Contents of letter to donors."""
+    return """Dear {},\nThank you for your very kind donation of {:.2f}.\n\nIt will be put to very good use.\n\n \t\tSincerely,\n\t\t\t-The Team""".format(
+        donor, donors[donor][-1])
+
+
+def send_letter_file():
+    """Write a thank you letter and save to file."""
+    # file_name = donor + '.txt'
+
+    for k, v in donors.items():
+        file_name = k + '.txt'
+        text = letter(k)
+        with open(file_name, 'w') as f:
+            f.write(text)
+
+    print('Completed creating letters to send out to donors.')
+
+
+def quit():
+    """This function quits the donation management system."""
+    return sys.exit('Exiting the system. Please wait...')
+
+
+def print_header():
+    """Prints the menu items to choose from and returns the selection."""
+
+    print('------------------------------------------')
+    print('       Donation Management System')
+    print('                 v0.1.3\n')
+    print('       1: Send A Thank You')
+    print('       2: Create A Report')
+    print('       3: Send Letters To Everyone')
+    print('       4: Quit\n')
+    print('------------------------------------------\n')
+
+    try:
+        selection = int(input('Please select a menu item: '))
+    except ValueError:
+        print('Your selection is invalid. Please make a selection from the menu.')
+        selection = int(input('Please select a menu item: '))
```

Comment:
what if they enter two invalid selects?  Perhaps you should wrap this in a `while`:
```
while True:
    try:
        selection = int(input('Please select a menu item: '))
    except ValueError:
        print('Your selection is invalid. Please make a selection from the menu.')
        continue